lib/base/io-engine.hpp

```diff
@@ -16,11 +16,16 @@
 #include <utility>
 #include <vector>
 #include <stdexcept>
+#include <boost/context/fixedsize_stack.hpp>
```

Comment:
I think that inclusion should also be guarded by the boost version check down below.